pandas/core/groupby.py

```diff
@@ -3531,6 +3531,13 @@ def transform(self, func, *args, **kwargs):
         Each subframe is endowed the attribute 'name' in case you need to know
         which group you are working on.
 
+        The current implementation imposes two requirements on f:
+
+        * f must return a new DataFrame - in-place mutation of subframes is
+          not supported and may produce unexpected results.
```

Comment:
This is not entirely correct. It can certainly return other things as well (like scalars), as long as it broadcasts to a Series or DataFrame of the same shape as the subset.